README.md

```diff
@@ -557,15 +557,20 @@ Note that the flattener does not deal with arrays at this time.
 
 By default, the fluentd elasticsearch plugin does not emit records with a _id field, leaving it to Elasticsearch to generate a unique _id as the record is indexed. When an Elasticsearch cluster is congested and begins to take longer to respond than the configured request_timeout, the fluentd elasticsearch plugin will re-send the same bulk request. Since Elasticsearch can't tell its actually the same request, all documents in the request are indexed again resulting in duplicate data. In certain scenarios, this can result in essentially and infinite loop generating multiple copies of the same data.
 
-Using an _id generated by the fluentd elasticsearch plugin will communicate to Elasticsearch the uniqueness of the requests so that duplicates will be rejected or simply replace the existing records.
+The bundled elasticsearch_genid filter can generate a unique _hash key for each record, this key may be passed to the id_key parameter in the elasticsearch plugin to communicate to Elasticsearch the uniqueness of the requests so that duplicates will be rejected or simply replace the existing records.
 Here is a sample config:
 
 ```
-<hash>
-  hash_id_key _id    # storing generated hash id key
-  hash_id_key _hash    # storing generated hash id key
-</hash>
-id_key _hash # specify same key name which is specified in hash_id_key
+<filter>
+  @type elasticsearch_genid
+  hash_id_key _hash    # storing generated hash id key (default is _hash)
+</filter>
+<match **>
+  @type elasticsearch
+  id_key _hash # specify same key name which is specified in hash_id_key
+  remove_keys _hash # Elasticsearch doesn't like keys that start with _
```

Comment:
> Elasticsearch doesn't like keys that start with _ ...

Hmm... is this true as a general statement? And BTW is there any official ES docs stating that keys starting with `_` are forbidden or anything?

Reply:
The message "Field names beginning with _ are not supported" apears next to the field in Kibana.:
![image](https://user-images.githubusercontent.com/24274156/33112063-3324c65a-cf1f-11e7-91e5-f45e66e0415c.png)

mmm.... perhaps it's a Kibana thing? not Elasticsearch per-se.

In general, do you see value in emitting both keys? seem redundant to me.

Reply:
Preserving `_hash` key in records seems redundant to me.
BTW, this is recommended configuration.
Users don't have to follow this configuration, but this configuration should be a good starting point.

Reply:
@dannyk81 I believe this is Kibana specific restriction only, see https://github.com/elastic/kibana/issues/14856 and linked issues from there...

I can imagine that ES itself could reject indexing a document if it contains `_id` field as it is either expected to be exclusively generated by ES (the [POST](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/docs-index_.html#_automatic_id_generation) operation) or the client is expected to provide the value as part of URL (the [PUT](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/docs-index_.html#docs-index_) operation). Nothing in between. However, general statement like that one in proposed code is not really correct IMO.

Reply:
Indeed, this is a Kibana issue not Elasticsearch, so I would suggest to re-phrase:
```
  remove_keys _hash # to avoid storing the hash in duplicate fields (optional)
```